tests/User/AutoLoginMiddlewareTest.php

```diff
@@ -0,0 +1,201 @@
+     */
+    private $requestHandlerMock;
+
+    /**
+     * @var ServerRequestInterface
+     */
+    private $requestMock;
+
+    /**
+     * @var AutoLoginMiddleware
+     */
+    private $autoLoginMiddlewareMock;
+
+    /**
+     * @var IdentityRepositoryInterface
+     */
+    private $identityRepositoryInterfaceMock;
+
+    /**
+     * @var IdentityInterface
+     */
+    private $identityInterfaceMock;
+
+    /**
+     * @var Logger
+     */
+    private $loggerMock;
+
+    /**
+     * @var User
+     */
+    private $userMock;
+
+    public function testProcessOK(): void
+    {
+        $this->mockDataRequest();
+        $this->mockDataCookie(["remember" => json_encode(['1', 'ABCD1234', 60])]);
+        $this->mockFindIdentity();
+
+        $this->userMock
+            ->expects($this->once())
+            ->method('validateAuthKey')
+            ->willReturn(true);
+
+        $this->userMock
+            ->expects($this->once())
+            ->method('login')
+            ->willReturn(true);
+
+        $response = new Response();
+        $this->requestHandlerMock
+            ->expects($this->once())
+            ->method('handle')
+            ->willReturn($response);
+
+        $this->assertEquals($this->autoLoginMiddlewareMock->process($this->requestMock, $this->requestHandlerMock), $response);
+    }
+
+    public function testProcessErrorLogin(): void
+    {
+        $this->mockDataRequest();
+        $this->mockDataCookie(["remember" => json_encode(['1', 'ABCD1234', 60])]);
+        $this->mockFindIdentity();
+
+        $this->userMock
+            ->expects($this->once())
+            ->method('validateAuthKey')
+            ->willReturn(true);
+
+        $this->userMock
+            ->expects($this->once())
+            ->method('login')
+            ->willReturn(false);
+
+        $memory = memory_get_usage();
+        $this->loggerMock->setTraceLevel(3);
+
+        $this->autoLoginMiddlewareMock->process($this->requestMock, $this->requestHandlerMock);
+
+        $messages = $this->getInaccessibleProperty($this->loggerMock, 'messages');
+        $this->assertEquals($messages[0][1], 'Unable to authenticate used by cookie.');
+    }
+
+    public function testProcessInvalidAuthKey(): void
+    {
+        $this->mockDataRequest();
+        $this->mockDataCookie(["remember" => json_encode(['1', '123456', 60])]);
+        $this->mockFindIdentity();
+
+        $memory = memory_get_usage();
+        $this->loggerMock->setTraceLevel(3);
+
+        $this->autoLoginMiddlewareMock->process($this->requestMock, $this->requestHandlerMock);
+
+        $messages = $this->getInaccessibleProperty($this->loggerMock, 'messages');
+        $this->assertEquals($messages[0][1], 'Unable to authenticate used by cookie. Invalid auth key.');
+    }
+
+    public function testProcessCookieEmpty(): void
+    {
+        $this->mockDataRequest();
+        $this->mockDataCookie([]);
+        $this->mockFindIdentity();
+
+        $memory = memory_get_usage();
```

Comment:
Should be removed. Please check other methods too

Reply:
I removed the mockFindIdentity. Is it ok?

Reply:
Yes, but why do you need this line?

Reply:
Is true, this line:

$memory = memory_get_usage();

Not is necessary. I updated the PR.